<!DOCTYPE html><html><head><meta charset="utf-8"><title>Untitled Document.md</title><style></style></head><body id="preview">
<h1><a id="CTT_DUPLICATED_GENE_ANNOTATION_0"></a>CTT DUPLICATED GENE ANNOTATION</h1>
<h4><a id="TEAM_NAME_PerlSquad_perlsquadougmailcom_1"></a>TEAM NAME: PerlSquad <a href="mailto:perlsquadou@gmail.com">perlsquadou@gmail.com</a></h4>
<ul>
<li>Alex Mayle <a href="mailto:am218112@ohio.edu">am218112@ohio.edu</a></li>
<li>Brian Reynolds <a href="mailto:br058215@ohio.edu">br058215@ohio.edu</a></li>
<li>Eric Keep <a href="mailto:ek349112@ohio.edu">ek349112@ohio.edu</a></li>
<li>Favour Ogundare <a href="mailto:fo893113@ohio.edu">fo893113@ohio.edu</a></li>
<li>Robert Smith <a href="mailto:rs659612@ohio.edu">rs659612@ohio.edu</a></li>
</ul>
<h4><a id="CLIENT_Zhihua_Hua_huaohioedu_8"></a>CLIENT: Zhihua Hua <a href="mailto:hua@ohio.edu">hua@ohio.edu</a></h4>
<pre><code>Availability: 6:30 - 7:00 pm Tuesday/Thursday
Preferred Method of Communication: E-mail
</code></pre>
<h4><a id="INITIAL_PROJECT_DESCRIPTION_12"></a>INITIAL PROJECT DESCRIPTION</h4>
<blockquote>
<p>What is the maximum number of genes encoded in each     genome?</p>
</blockquote>
<p>Students work to answer this question by ‘finalizing’ a PERL-based annotation code, &gt;that implements a closing Target Trimming algorithm, to &gt;re-annotate the missing genes in their preferred genomes.</p>
<h4><a id="INTENDED_USER_BASE_18"></a>INTENDED USER BASE</h4>
<p>Academia, Bioinformatics Personnel.</p>
<h4><a id="FIRST_ADOPTER_USER_GROUP_21"></a>FIRST ADOPTER USER GROUP</h4>
<p>Academia, Bioinformatics Personnel.</p>
<h4><a id="EQUIPMENT_NEEDS_24"></a>EQUIPMENT NEEDS</h4>
<pre><code>Machine running Perl with required dependencies:
    Bioperl, BioPro, GeneWIse, Blast, HMMER

Current Requirements:
    - Server to compile resulting executables,          
    - Provided by Dr. Hua. 

    - Create account on server so students can access files on server.

    - Tools/Server for Web-Site for developed stand alone program. 

    - Eventually, student research assistants along with other 
      professors for alpha/beta testing 
</code></pre>
<h4><a id="FEATURES_40"></a>FEATURES</h4>
<blockquote>
<p>Executable File:- specific to eukaryotic systems :- application is not web based</p>
</blockquote>
<ul>
<li>
<p>Application takes input from user in the firm of annotation file, sequence of proteins and family name.</p>
</li>
<li>
<p>Application finds genome, then finds where previous annotation file missed genes and or potential new genes, translate new signs, using dependency GeneWise to use previous annotated sequences.</p>
</li>
<li>
<dl>
<dt>Application outputs how many genes belong to input’s family.</dt>
<dd>Genomic DNA Sequence, Annotation Transcript sequence, Peptide Sequence</dd>
</dl>
</li>
</ul>
<h4><a id="INITIAL_USE_CASES__USAGE_SCENARIOS_50"></a>INITIAL USE CASES / USAGE SCENARIOS</h4>
<blockquote>
<p>To annotate duplicated genes that may have been missed in the initial .
Currently not enough information for more specific use cases and scenarios,
Section will be updated by the next time the Team meets with the client.</p>
</blockquote>
<h4><a id="NONFUNCTIONAL_REQUIREMENTS_55"></a>NON-FUNCTIONAL REQUIREMENTS</h4>
<blockquote>
<p>Security is not a priority, all code used will have open source propiety rights
Dependencies and requirements for the program will be listed on the eventual site
and the executables FAQ.</p>
</blockquote>

</body></html>